### PR TITLE
PUBDEV-7720: RuleFit tests and docs adjustments

### PR DESCRIPTION
**Pull request by koniecsveta**
https://0xdata.atlassian.net/browse/PUBDEV-7720

### CODE REVIEW

**Comment by ledell** [comment]
I have another PR over here covering some related RuleFit user guide pages... I put it into rel-zeno since that's where the original PR went. https://github.com/h2oai/h2o-3/pull/4943

File: h2o-bindings/bin/custom/R/gen_rulefit.py [hunk 1]
```diff
@@ -38,7 +40,7 @@
 predictors <- c("age", "sibsp", "parch", "fare", "sex", "pclass")
 titanic[,response] <- as.factor(titanic[,response])
 titanic[,"pclass"] <- as.factor(titanic[,"pclass"])
-rf_h2o = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10, 
+rfit = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10,
```
Comment (ledell) [inline h2o-bindings/bin/custom/R/gen_rulefit.py:43]:
```suggestion
rfit <- h2o.rulefit(y = response, x = predictors, training_frame = titanic, max_rule_length = 10, 
```

File: h2o-bindings/bin/custom/R/gen_rulefit.py [hunk 2]
```diff
@@ -38,7 +40,7 @@
 predictors <- c("age", "sibsp", "parch", "fare", "sex", "pclass")
 titanic[,response] <- as.factor(titanic[,response])
 titanic[,"pclass"] <- as.factor(titanic[,"pclass"])
-rf_h2o = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10, 
+rfit = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10, 
 max_num_rules=100, seed=1234, model_type="rules")
```
Comment (ledell) [inline h2o-bindings/bin/custom/R/gen_rulefit.py:44]:
```suggestion
max_num_rules = 100, seed = 1, model_type = "rules")
```

File: h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R [hunk 3]
```diff
@@ -15,12 +15,20 @@ test.rulefit.titanic <- function() {
     titanic[,response] <- as.factor(titanic[,response])
     titanic[,"pclass"] <- as.factor(titanic[,"pclass"])
 
-    rf_h2o = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10, max_num_rules=100, seed=1234, model_type="rules")
+    # Split the dataset into train and test
+    splits <- h2o.splitFrame(data = titanic, ratios = .8, seed = 1234)
+    train <- splits[[1]]
+    test <- splits[[2]]
 
+    rf_h2o = h2o.rulefit(y=response, x=predictors, training_frame = train, max_rule_length=10, max_num_rules=100, seed=1234, model_type="rules")
```
Comment (ledell) [inline h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R:23]:
```suggestion
    rfit <- h2o.rulefit(y = response, x = predictors, training_frame = train, max_rule_length = 10, max_num_rules = 100, seed = 1, model_type="rules")
```

File: h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R [hunk 4]
```diff
@@ -15,12 +15,20 @@ test.rulefit.titanic <- function() {
...
+
+    print(rf_h2o@model$rule_importance)
```
Comment (ledell) [inline h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R:25]:
```suggestion
    print(rfit@model$rule_importance)
```

File: h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R [hunk 5]
```diff
@@ -15,12 +15,20 @@ test.rulefit.titanic <- function() {
...
+
+    h2o.predict(rf_h2o, newdata=test)
```
Comment (ledell) [inline h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R:27]:
```suggestion
    h2o.predict(rfit, newdata = test)
```

File: h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R [hunk 6]
```diff
@@ -15,12 +15,20 @@ test.rulefit.titanic <- function() {
...
     print(rf_h2o@model$rule_importance)
 
-    expect_that(h2o.auc(h2o.performance(rf_h2o)), equals(h2o.auc(h2o.performance(rf_h2o, newdata =  titanic))))
-    expect_that(h2o.logloss(h2o.performance(rf_h2o)), equals(h2o.logloss(h2o.performance(rf_h2o, newdata =  titanic))))
+    expect_that(h2o.auc(h2o.performance(rf_h2o)), equals(h2o.auc(h2o.performance(rf_h2o, newdata =  test))))
```
Comment (ledell) [inline h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R:30]:
```suggestion
    expect_that(h2o.auc(h2o.performance(rfit)), equals(h2o.auc(h2o.performance(rfit, newdata =  test))))
```

File: h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R [hunk 7]
```diff
@@ -15,12 +15,20 @@ test.rulefit.titanic <- function() {
...
+    expect_that(h2o.logloss(h2o.performance(rf_h2o)), equals(h2o.logloss(h2o.performance(rf_h2o, newdata =  test))))
```
Comment (ledell) [inline h2o-r/tests/testdir_algos/rulefit/runit_rulefit_titanic.R:31]:
```suggestion
    expect_that(h2o.logloss(h2o.performance(rfit)), equals(h2o.logloss(h2o.performance(rfit, newdata =  test))))
```

File: h2o-r/h2o-package/R/rulefit.R [hunk 8]
```diff
@@ -42,7 +42,7 @@
 #' predictors <- c("age", "sibsp", "parch", "fare", "sex", "pclass")
 #' titanic[,response] <- as.factor(titanic[,response])
 #' titanic[,"pclass"] <- as.factor(titanic[,"pclass"])
-#' rf_h2o = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10, 
+#' rfit = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10,
```
Comment (ledell) [inline h2o-r/h2o-package/R/rulefit.R:45]:
```suggestion
#' rfit <- h2o.rulefit(y = response, x = predictors, training_frame = titanic, max_rule_length = 10, 
```

File: h2o-r/h2o-package/R/rulefit.R [hunk 9]
```diff
@@ -42,7 +42,7 @@
 #' predictors <- c("age", "sibsp", "parch", "fare", "sex", "pclass")
 #' titanic[,response] <- as.factor(titanic[,response])
 #' titanic[,"pclass"] <- as.factor(titanic[,"pclass"])
-#' rf_h2o = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10, 
+#' rfit = h2o.rulefit(y=response, x=predictors, training_frame = titanic, max_rule_length=10, 
 #' max_num_rules=100, seed=1234, model_type="rules")
```
Comment (ledell) [inline h2o-r/h2o-package/R/rulefit.R:46]:
```suggestion
#' max_num_rules = 100, seed=1, model_type = "rules")
```

**Review by ledell — CHANGES_REQUESTED** [review]
I made a few commit suggestions for R style guide alignment.  I am also wondering if you can update the example in the Rdocs (created in `gen_rulefit.py`) to use the example here (I made some edits to the user guide rst file): https://github.com/h2oai/h2o-3/blob/7268ddecf200cd11f73cb743efd93ca50089fd28/h2o-docs/src/product/data-science/rulefit.rst#examples

**Comment by koniecsveta** [comment]
> I made a few commit suggestions for R style guide alignment. I am also wondering if you can update the example in the Rdocs (created in `gen_rulefit.py`) to use the example here for consistency -- there are a few slight changes (I made some edits to the user guide rst file): https://github.com/h2oai/h2o-3/blob/7268ddecf200cd11f73cb743efd93ca50089fd28/h2o-docs/src/product/data-science/rulefit.rst#examples


3cf6c7a

**Review by ledell — APPROVED** [review]
This is great, thanks!

**Comment by koniecsveta** [comment]
added do not merge as tests are failing / need to investigate why

**Pull request merged**